.github/workflows/actions.yml

```diff
@@ -25,9 +25,29 @@ jobs:
       - name: Test
         run: yarn test
 
-  build-and-deploy:
+  build:
     runs-on: ubuntu-latest
     needs: test
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+
+      - name: Install Yarn and dependencies
+        run: |
+          npm install -g yarn
+          yarn install
+
+      - name: Build
+        run: yarn build

```

Comment:
Do we need to `yarn install` for every job if we have installed it in the previous job? And the same for `build`, do we need it in `deploy` again if we have before in the `build` job

Reply:
yes, because every job starts on a separate virtual machine